timeouts/type/timeouts.go

```diff
@@ -0,0 +1,120 @@
+package timeouts
```

Comment:
I wonder if this should be promoted to the top level `timeouts` package so provider developers do not have an awkward `type` import if they do not import alias it?